core/src/main/java/org/apache/accumulo/core/client/impl/ClientConfConverter.java

```diff
@@ -0,0 +1,105 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.client.impl;
+
+import java.util.Iterator;
+import java.util.Properties;
+
+import org.apache.accumulo.core.client.ClientConfiguration;
+import org.apache.accumulo.core.conf.ClientProperty;
+
+public class ClientConfConverter {
+
+  public static ClientConfiguration toClientConf(Properties properties) {
+    ClientConfiguration config = ClientConfiguration.create();
+    for (Object keyObj : properties.keySet()) {
+      String key = (String) keyObj;
+      String val = properties.getProperty(key);
+      if (key.equals(ClientProperty.INSTANCE_ZOOKEEPERS.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.INSTANCE_ZK_HOST, val);
+      } else if (key.equals(ClientProperty.INSTANCE_ZOOKEEPERS_TIMEOUT_SEC.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.INSTANCE_ZK_TIMEOUT, val);
+      } else if (key.equals(ClientProperty.SSL_ENABLED.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.INSTANCE_RPC_SSL_ENABLED, val);
+      } else if (key.equals(ClientProperty.SSL_KEYSTORE_PATH.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SSL_KEYSTORE_PATH, val);
+        config.setProperty(ClientConfiguration.ClientProperty.INSTANCE_RPC_SSL_CLIENT_AUTH, "true");
+      } else if (key.equals(ClientProperty.SSL_KEYSTORE_TYPE.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SSL_KEYSTORE_TYPE, val);
+      } else if (key.equals(ClientProperty.SSL_KEYSTORE_PASSWORD.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SSL_KEYSTORE_PASSWORD, val);
+      } else if (key.equals(ClientProperty.SSL_TRUSTSTORE_PATH.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SSL_TRUSTSTORE_PATH, val);
+      } else if (key.equals(ClientProperty.SSL_TRUSTSTORE_TYPE.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SSL_TRUSTSTORE_TYPE, val);
+      } else if (key.equals(ClientProperty.SSL_TRUSTSTORE_PASSWORD.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SSL_TRUSTSTORE_PASSWORD, val);
+      } else if (key.equals(ClientProperty.SSL_USE_JSSE.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_USE_JSSE, val);
+      } else if (key.equals(ClientProperty.SASL_ENABLED.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.INSTANCE_RPC_SASL_ENABLED, val);
+      } else if (key.equals(ClientProperty.SASL_QOP.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.RPC_SASL_QOP, val);
+      } else if (key.equals(ClientProperty.SASL_KERBEROS_SERVER_PRIMARY.getKey())) {
+        config.setProperty(ClientConfiguration.ClientProperty.KERBEROS_SERVER_PRIMARY, val);
+      } else {
+        config.setProperty(key, val);
```

Comment:
Switch case might be cleaner

Reply:
I think they both are kind of ugly.  I would prefer to keep it as is.